src/backend/Controllers/UserController.cs

```diff
@@ -20,20 +21,28 @@ public class UserController : Controller
     {
         private readonly IMapper _mapper;
         private readonly UsersService _usersService;
+        private readonly FileUploadService _fileUploadService;
 
-        public UserController(IMapper mapper, UsersService usersService)
+        public UserController(IMapper mapper, UsersService usersService, FileUploadService fileUploadService)
         {
             _mapper = mapper;
             _usersService = usersService;
+            _fileUploadService = fileUploadService;
         }
 
         [HttpPost("register")] // "api/user/register"
-        public ActionResult<User> Register(CreateUserDto createUserDto)
+        public async Task<ActionResult<User>> RegisterAsync([FromForm] CreateUserDto createUserDto)
         {
             if (_usersService.GetByEmail(createUserDto.Email) != null)
                 return Conflict("User with the same email already exists");
-            
+
+            var avatarPath = await _fileUploadService.UploadFormFileAsync(createUserDto.Avatar, "images");
+
+            if (avatarPath == null)
+                return BadRequest("Invalid image file");
```

Comment:
Should we force people to use avatars? Maybe we should use some kind of generic avatar for people who do not provide avatars?

Reply:
I have mentioned to @KungFur1 that avatars should be optional but he apparently forgot 😞